rclcpp/src/rclcpp/executable_list.cpp

```diff
@@ -24,3 +24,57 @@ ExecutableList::ExecutableList()
 
 ExecutableList::~ExecutableList()
 {}
+
+void
+ExecutableList::clear()
+{
+  this->timer.clear();
+  this->number_of_timers = 0;
+
+  this->subscription.clear();
+  this->number_of_subscriptions = 0;
+
+  this->service.clear();
+  this->number_of_services = 0;
+
+  this->client.clear();
+  this->number_of_clients = 0;
+
+  this->waitable.clear();
+  this->number_of_waitables = 0;
+}
+
+void
+ExecutableList::add_subscription(rclcpp::SubscriptionBase::SharedPtr subscription)
+{
+  this->subscription.push_back(subscription);
```

Comment:
use `std::move(subscription)` here and in all the following functions.

`std::move` with shared pointer is useful when you only need to pass the pointer to someone else, as it does not increase the atomic counter for decreasing it immediately after